examples/stable-diffusion/image_to_video_generation.py

```diff
@@ -188,41 +230,68 @@ def main():
     if args.bf16:
         kwargs["torch_dtype"] = torch.bfloat16
 
-    pipeline = GaudiStableVideoDiffusionPipeline.from_pretrained(
-        args.model_name_or_path,
-        **kwargs,
-    )
+    if args.control_image_path is not None:
+        from optimum.habana.diffusers import GaudiStableVideoDiffusionPipelineControlNet
+        from optimum.habana.diffusers.models import ControlNetSDVModel
+        from optimum.habana.diffusers.models import UNetSpatioTemporalConditionControlNetModel
+        controlnet = controlnet = ControlNetSDVModel.from_pretrained(
+            args.controlnet_model_name_or_path,
+            subfolder="controlnet")
+        unet = UNetSpatioTemporalConditionControlNetModel.from_pretrained(
+            args.model_name_or_path,
+            subfolder="unet")
+        pipeline = GaudiStableVideoDiffusionPipelineControlNet.from_pretrained(
+            args.model_name_or_path,
+            controlnet=controlnet,
+            unet=unet,
+            **kwargs)
 
-    # Set seed before running the model
-    set_seed(args.seed)
+        # Set seed before running the model
+        set_seed(args.seed)
```

Comment:
```suggestion
```

Isn't it better to set the random seed before loading the models before the conditional? 
Personal suggestion! I think this ensures that any randomness involved in the model loading process (such as weight initialization for certain layers) is controlled, leading to reproducible results.

Reply:
Done, Thanks.